Automattic/wp-calypso
# [POC] Pattern Assembler: Support block preview on calypso

#### Proposed Changes

**DON'T MERGE THIS PR** 

Try to render the blocks and styles on the client side. There are several issues we need to solve
* dependencies issues: The same package might be installed multiple times and we might use all of them on calypso and it leads that some global status don't work. Workaround: define resolutions in package.json
  * Case 1: [blockPropsProvider and innerBlocksPropsProvider](https://github.com/WordPress/gutenberg/blob/ff13a7e4572c19eca9747f7f044386099cc60cb8/packages/blocks/src/api/serializer.js#L74) are relied on the global state, so it won't work if we install multiple versions.
  * Case 2: [Hooks](https://github.com/WordPress/gutenberg/blob/ff13a7e4572c19eca9747f7f044386099cc60cb8/packages/hooks/src/createHooks.js#L19) are also relied on the global state
* api-fetch: we have apiFetch proxy via `gutenberg-wpcom-apifetch.js` in the editor, and we need to have the same mechanism on calypso. Workaround: copy and re-write `gutenberg-wpcom-apifetch.js` on calypso 
* styles: I just verify several patterns and I cannot guarantee all of the styles work as expected
* components: some Gutenberg components are not exported well and they might not easy to use. So, maybe we could just implement a simple version on the calypso side and we don't need to import lots of dependencies.

https://user-images.githubusercontent.com/13596067/187589985-27914f8d-125b-4a79-b23b-ef2fc21dd32f.mov

#### Testing Instructions



* Apply D86835-code to your sandbox
* Go to `/themes/<your_site>` and select any theme with style variations. ex. Pedant
* Go to `/setup/patternAssembler?siteSlug=<your_site>`
* Select any pattern, it have to be rendered into right side
* Select any style variations, you have to see the style is changed on both left and right side

#### Pre-merge Checklist






Related to https://github.com/Automattic/wp-calypso/issues/66589
